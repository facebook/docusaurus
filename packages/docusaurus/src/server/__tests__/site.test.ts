/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import {mergeWithCustomize} from 'webpack-merge';
import {loadSetup} from './testUtils';
import type {Props} from '@docusaurus/types';
import type {DeepPartial} from 'utility-types';

describe('load', () => {
  it('loads props for site with custom i18n path', async () => {
    const site = await loadSetup('custom-i18n-site');
    expect(site.props).toMatchSnapshot();
    const site2 = await loadSetup('custom-i18n-site', {locale: 'zh-Hans'});
    expect(site2.props).toEqual(
      mergeWithCustomize<DeepPartial<Props>>({
        customizeArray(a, b, key) {
          return ['routesPaths', 'plugins'].includes(key) ? b : undefined;
        },
      })(site.props, {
        baseUrl: '/zh-Hans/',
        i18n: {
          currentLocale: 'zh-Hans',
        },
        localizationDir: path.join(
          __dirname,
          '__fixtures__/custom-i18n-site/i18n/zh-Hans-custom',
        ),
        outDir: path.join(
          __dirname,
          '__fixtures__/custom-i18n-site/build/zh-Hans',
        ),
        routesPaths: ['/zh-Hans/404.html'],
        siteConfig: {
          baseUrl: '/zh-Hans/',
        },
        siteStorage: {
          namespace: '',
          type: 'localStorage',
        },
        plugins: site2.props.plugins,
      }),
    );
  });
});
