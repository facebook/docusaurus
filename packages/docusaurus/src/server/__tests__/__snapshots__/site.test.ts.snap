// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load loads props for site with custom i18n path 1`] = `
{
  "baseUrl": "/",
  "codeTranslations": {},
  "generatedFilesDir": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site/.docusaurus",
  "headTags": "",
  "i18n": {
    "currentLocale": "en",
    "defaultLocale": "en",
    "localeConfigs": {
      "en": {
        "calendar": "gregory",
        "direction": "ltr",
        "htmlLang": "en",
        "label": "English",
        "path": "en-custom",
      },
      "zh-Hans": {
        "calendar": "gregory",
        "direction": "ltr",
        "htmlLang": "zh-Hans",
        "label": "简体中文",
        "path": "zh-Hans-custom",
      },
    },
    "locales": [
      "en",
      "zh-Hans",
    ],
    "path": "i18n",
  },
  "localizationDir": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site/i18n/en-custom",
  "outDir": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site/build",
  "plugins": [
    {
      "content": undefined,
      "defaultCodeTranslations": {},
      "getClientModules": [Function],
      "globalData": undefined,
      "injectHtmlTags": [Function],
      "name": "docusaurus-bootstrap-plugin",
      "options": {
        "id": "default",
      },
      "path": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site",
      "routes": [],
      "version": {
        "type": "synthetic",
      },
    },
    {
      "configureWebpack": [Function],
      "content": undefined,
      "defaultCodeTranslations": {},
      "globalData": undefined,
      "name": "docusaurus-mdx-fallback-plugin",
      "options": {
        "id": "default",
      },
      "path": ".",
      "routes": [],
      "version": {
        "type": "synthetic",
      },
    },
  ],
  "postBodyTags": "",
  "preBodyTags": "",
  "routes": [],
  "routesPaths": [
    "/404.html",
  ],
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {
        "en": {
          "direction": "ltr",
          "path": "en-custom",
        },
        "zh-Hans": {
          "direction": "ltr",
          "path": "zh-Hans-custom",
        },
      },
      "locales": [
        "en",
        "zh-Hans",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "Site",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site/docusaurus.config.js",
  "siteDir": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/custom-i18n-site",
  "siteMetadata": {
    "docusaurusVersion": "<CURRENT_VERSION>",
    "pluginVersions": {},
    "siteVersion": undefined,
  },
  "siteStorage": {
    "namespace": "",
    "type": "localStorage",
  },
  "siteVersion": undefined,
}
`;
